dsgrid/registry/project_registry_manager.py

```diff
@@ -823,11 +827,22 @@ def _submit_dataset(
         if os.environ.get("__DSGRID_SKIP_DATASET_TO_PROJECT_MAPPING_CHECKS__") is not None:
             logger.warning("Skip dataset-to-project mapping checks")
         else:
-            self._check_dataset_base_to_project_base_mappings(
-                project_config,
-                dataset_config,
-                mapping_references,
-            )
+            # This operation can be very problematic if there are many executors and runs much
+            # faster with a single executor on a single core.
+            # The dynamic allocation settings will only take effect if the worker was started
+            # with spark.shuffle.service.enabled=true
+            conf = {
+                "spark.dynamicAllocation.enabled": True,
+                "spark.dynamicAllocation.shuffleTracking.enabled": True,
+                "spark.dynamicAllocation.maxExecutors": 1,
+                "spark.executor.cores": 1,
+            }
+            with restart_spark_with_custom_conf(conf):
+                self._check_dataset_base_to_project_base_mappings(
+                    project_config,
+                    dataset_config,
+                    mapping_references,
+                )
```

Comment:
What happens when this exits? Does it somehow go back to the previous spark config?

Reply:
I see now that it does. I wasn't reading function names carefully enough ...